File: TravelPlannerClient/src/app/model/trip.ts
export class Trip{
   
    id:number;
    destination:string;
    startDate:string;
    endDate:string;
    comment:string;
    sightseeings:string[];
    lng:number;
    lat:number;
    sights:string;
    userEmail:string;
    counter:string;
    
    constructor(private object?:any) {
        if(object){
        this.id=object.id;
        this.destination=object.destination;
        this.startDate=new Date(object.startDate).toLocaleDateString();
        this.endDate=new Date(object.endDate).toLocaleDateString();
        this.comment=object.comment;
        this.sightseeings=object.sightseeings;
        this.sights = object.sightseeings.join(", ")
        this.lng=object.lng;
        this.lat=object.lat;
        this.userEmail = object.userEmail;
        this.counter = this.getCounter(object.startDate);
        }
        }

        getCounter(startDate): string {
            let oneDay = 24*60*60*1000; // hours*minutes*seconds*milliseconds
            let today = new Date();
            let start = new Date(startDate);
            if(start < today)
                return 'Already Started' 
            let diffDays = Math.floor(Math.abs((today.getTime() - start.getTime())/(oneDay)))+1;
            return diffDays.toString()+' days left';
        }
    }

File: TravelPlanner.Shared/Entities/Trip.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TravelPlanner.Shared.Entities
{
    public class Trip
    {
        public int Id { get; set; }

        public string Destination { get; set; }

        public string Comment { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public double Lat { get; set; }

        public double Lng { get; set; }

        public string SightseeingsCollection { get; set; }

        public string TravelUserId { get; set; }

        public TravelUser TravelUser { get; set; }
    }
}
